### PR TITLE
add Website

### PR DESCRIPTION
**Pull request by naskio**
Initial release of the G-Research organization website, featuring:

1. UI built with Gatsby, TypeScript, and Mantine UI, including:
   - Hero section with a brief introduction to G-Research's open-source software role
   - Featured projects section showcasing selected projects
   - All projects section displaying G-Research's open-source projects with filtering and sorting capabilities

2. Automated workflow that:
   - Fetches and processes project data from the GitHub API, including:
     • Project details (name, stars, forks, description, etc.)
     • Community URLs (Discord, Slack, etc.) extracted from each project's `README.md`
   - Builds the website
   - Deploys to GitHub Pages
   - Runs every few hours

3. More features:
   - Project detail overrides
   - Custom project logos
   - Featured projects definition and ordering
   - Multi-organization data fetching (For example: G-Research, armadaproject, m4rs-mt)
   - Project exclusion from the website
   - Caching for: GitHub API, Gatsby build step, and more.

This release provides a comprehensive showcase of G-Research's open-source initiatives with automated data collection and website deployment.

### CODE REVIEW

File: README.md
Comment (stackedsax) [inline README.md:6]: I'd like to wordsmith this a little.  It's not bad, I just think it could be better.

File: src/components/IndexPage/HeroSection/HeroSection.view.tsx
Comment (stackedsax) [inline src/components/IndexPage/HeroSection/HeroSection.view.tsx:25]: Maybe two <br />'s here?  Somehow the single line break looks weird to me.

Comment (stackedsax) [reply src/components/IndexPage/HeroSection/HeroSection.view.tsx:25]: hahah, of course github interpreted my "br"'s in that previous comment.  Genius.

**Review by stackedsax — CHANGES_REQUESTED** [review]
Overall looks great.

We should probably take out these repo's though:
 * .GITHUB
 * Brand
 * TIGPUBLIC

Also we should discuss:
* PARQUETSHARP.DATASET
* CRIO-FLATCAR-BUILD (we've moved away from Flatcar as a company)
* PARQUETSHARP.DATASET

And some of the lesser-known repo's down the list should be up for discussion whether to include them or not.

Eh, we should just remove the Bearcat repo itself, I think.  But all of that discussion can be done after you merge this in.  I mainly just think those first three repo's should be removed from the list for sure unless you have a compelling reason why not.

File: README.md
Comment (naskio) [reply README.md]: What do you think about the new description?

File: src/components/IndexPage/HeroSection/HeroSection.view.tsx
Comment (naskio) [reply src/components/IndexPage/HeroSection/HeroSection.view.tsx:25]: I added another br tag 👌

**Comment by naskio** [comment]
@stackedsax 
- I excluded these repos:  `G-Research/.github`, `G-Research/brand`, `G-Research/tigpublic`.
- For the 3rd one, I guess you mean `G-Research/bearcat` ?  It should not be a problem, once we decide we will just need to add them to the exclude list and they won't appear in the website.
- 

> could you squash all of the many commits and write a bit more about what's changing and why?

I am not sure what to do here, I expect that we will `Squash and Merge` this PR once we make the necessary changes, we will have 1 resulting commit that includes mainly: the website + CI/CD that collect the data, build and deploy it to GitHub Pages.

**Comment by stackedsax** [comment]
> I am not sure what to do here, I expect that we will Squash and Merge this PR once we make the necessary changes, we will have 1 resulting commit that includes mainly: the website + CI/CD that collect the data, build and deploy it to GitHub Pages.

That's fine as long as it gets cleaned up at that point. 

My comment about "what's changing and why" is more that these notes were pretty brief on what this PR was about:

> * Website
> * CI/CD to collect repos data automatically

I mean, I knew because I've been talking about it with you or Caterina, but someone else might not have immediately intuited everything that you were doing.  Just a couple more lines of explanation would be nice when you clean this up.

Lastly, I did get these warnings when I ran `yarn develop`:

```
 ERROR  UNKNOWN

(node:71093) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
```

**Comment by naskio** [comment]
@stackedsax 
- Understood, I will try to edit the PR description to make it more expilicit.
- For the warnings, it's related to the NodeJS version you are using (You won't see these warnings if you switch to the current LTS version == v20.15)

**Comment by stackedsax** [comment]
Thanks!  LGTM now

**Review by stackedsax — APPROVED** [review]
LGTM after your recent changes.  👍

**Pull request merged**
